walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -64,22 +62,50 @@
     Collection<String> attendees = new ArrayList<>();
     attendees.addAll(request.getAttendees());
 
+    // Create a list of all attendees (optional and mandatory) for the requested meeting
+    Collection<String> mandatoryAndOptionalAttendees = new ArrayList<>();
+    mandatoryAndOptionalAttendees.addAll(request.getOptionalAttendees());
+    mandatoryAndOptionalAttendees.addAll(attendees);
+
     // Collection of timeranges that can not work for the proposed schedule
     // These are times when the attendees have an event scheduled at that particular time
     List<TimeRange> attendeesCannotScheduleHere = cannotSheduleMeeting(attendees, events); 
 
-    // Return the whole day if all attendees are free for the whole day
-    if (attendeesCannotScheduleHere.isEmpty()) 
+    // Get the timeranges for the optional attendees
+    List<TimeRange>  mandatoryAndOptionalAttendeesCannotScheduleHere = cannotSheduleMeeting(mandatoryAndOptionalAttendees, events);
+
+    // Return the whole day if all attendees are free for the whole day annd there are no optional attendees
+    if (mandatoryAndOptionalAttendeesCannotScheduleHere.isEmpty()) 
       return Arrays.asList(TimeRange.WHOLE_DAY);
 
     // Sort the time ranges by the end time in ascending order
     Collections.sort(attendeesCannotScheduleHere, TimeRange.ORDER_BY_END);
+    Collections.sort(mandatoryAndOptionalAttendeesCannotScheduleHere, TimeRange.ORDER_BY_END);
 
     // Remove the time ranges which are part of longer time ranges
     removeNestedTimes(attendeesCannotScheduleHere);
+    removeNestedTimes(mandatoryAndOptionalAttendeesCannotScheduleHere);
 
-    for (int index = 0; index < attendeesCannotScheduleHere.size(); index++) {
-      TimeRange currentTimeRange = attendeesCannotScheduleHere.get(index);
+    // Check times that work for both optional and mandatory attendees
+    addRequestedMeetingToList(mandatoryAndOptionalAttendeesCannotScheduleHere, durationOfMeeting, possibleTimes);
+    // If there is no time that works for both optional and mandatory attendees, check for only mandatory attendees
+    if(possibleTimes.isEmpty()) {
+      addRequestedMeetingToList(attendeesCannotScheduleHere, durationOfMeeting, possibleTimes);
+    }
+    return possibleTimes;
+  }
+  /**
+   * Method runs the algorithm for adding possible times which work for either both mandatory and optional attendees or just mandatory attendees
+   * @param attendees is a list of all the attendees for the meeeting that we are considering
+   * @param durationOfMeeting is the length / duration of the requested meeting
+   * @param possibleTimes this is a list of all the possible times that we can schedule the meeting
+   */
+  private void addRequestedMeetingToList(List<TimeRange> attendees, long durationOfMeeting, Collection<TimeRange> possibleTimes) {
```

Comment:
Looks like `attendees` is now a list of time ranges, can we update the naming here to reflect that?